Signal-Windows/Controls/Conversation.xaml.cs

```diff
@@ -304,8 +312,12 @@ public void HandleDeleteMesage(SignalMessage message)
 
         private async void UserInputBar_OnEnterKeyPressed()
         {
-            bool shift = CoreWindow.GetForCurrentThread().GetKeyState(VirtualKey.Shift).HasFlag(CoreVirtualKeyStates.Down);
-            if (shift)
+            CoreWindow corWindow = CoreWindow.GetForCurrentThread();
+            bool shift = corWindow.GetKeyState(VirtualKey.Shift).HasFlag(CoreVirtualKeyStates.Down);
+            bool control = corWindow.GetKeyState(VirtualKey.Control).HasFlag(CoreVirtualKeyStates.Down);
+            // If SendMessageWithEnterEnabled is true then add new if shift key is pressed.
+            // If SendMessageWithEnterEnabled is false then send message if control key is pressed.
+            if (SendMessageWithEnterEnabled ? shift : !control)

```

Comment:
On desktop if you change the setting after having opened a conversation this flag does not get updated on this page because the Conversation.xaml does not get loaded again. I recommend just referring to the setting directly.

```suggestion
            if (GlobalSettingsManager.SendMessageWithEnterSetting ? shift : !control)
```